File: consumer.go
package main

import (
	"fmt"
	"strconv"
	"strings"

	amqp "github.com/rabbitmq/amqp091-go"
)

var username string = "guest"
var password string = "guest"
var host string = "localhost"
var port int = 5672

var queue_name string = "my-queue"

func main() {

	// connect to rabbitMQ
	connect_string := "amqp://" + username + ":" + password + "@" + host + ":" + strconv.Itoa(port)
	connection, _ := amqp.Dial(connect_string)

	// open a channel
	channel, _ := connection.Channel()

	msgs, _ := channel.Consume(
		queue_name, // queue string
		"",         // consumer string
		true,       // autoack
		false,      // exclusive
		false,      // no local
		false,      // no wait
		nil,        // args
	)

	forever := make(chan string)

	// go routine to handle messages
	go func(c chan string) {

		for d := range msgs {

			message := string(d.Body)

			fmt.Printf("msg: %s", message)

			if strings.HasPrefix(message, "stop") {
				break
			}
		}

		channel.Cancel("", false)
		connection.Close()

		c <- "done"

	}(forever)

	fmt.Println("Waiting for messages, send stop message to exit")

	// exit only when routine is done
	<-forever
}

File: consumer.py



import pika


host = 'localhost'
port = 5672

exchange_name = "my-exchange"
queue_name = "my-queue"

# connect to RabbitMQ server and create a channel
connection_params = pika.ConnectionParameters(host=host, port=port)
connection = pika.BlockingConnection(connection_params)

channel = connection.channel()

# define callback for whenever a message is available in the queue we are about to subscribe to
def callback(ch, method, properties, body):

	message = body.decode()

	if message.lower() == "stop":
		channel.stop_consuming()
		channel.close()
	
	print(message)


# consume messages sent to queue
print("Waiting for messages. Send `stop` or press `ctrl + c` to exit")

channel.basic_consume(queue=queue_name, on_message_callback=callback, auto_ack=True)
channel.start_consuming()